Running sample code in README

Sorry, I'm very new to Rust but need to replace `uchardet` with this `chardet` in a program to get it to compile on an ARM device.

I'm trying to run a project which is just the README code from this project:

**Cargo.toml:**

```
[package]

name = "hello_world"
version = "0.0.1"
authors = [ "Your name <you@example.com>" ]

[dependencies]
encoding = "0.2"
chardet = "0.2.4"

```

**src/main.rs:**

```
extern crate chardet;
extern crate encoding;
use chardet;
use std::fs::OpenOptions;
use std::io::prelude::*;
use encoding::DecoderTrap;
use encoding::label::encoding_from_whatwg_label;

// open text file
let mut fh = OpenOptions::new().read(true).open(filepath).expect(
    "Could not open file",
);
let mut reader: Vec<u8> = Vec::new();

// read file
fh.read_to_end(&mut reader).expect("Could not read file");

// detect charset of the file
let result = detect(&reader);
// result.0 Encode
// result.1 Confidence
// result.2 Language

// decode file into utf-8
let coder = encoding_from_whatwg_label(charset2encoding(&result.0));
if coder.is_some() {
    let utf8reader = coder.unwrap().decode(&reader, DecoderTrap::Ignore).expect("Error");
}
```

I get back:

```
   Compiling hello_world v0.0.1 (file:///tmp/hello)
error: expected item, found `let`
  --> src/main.rs:10:1
   |
10 | let mut fh = OpenOptions::new().read(true).open(filepath).expect(
   | ^^^

error: aborting due to previous error

error: Could not compile `hello_world`.

To learn more, run the command again with --verbose.

```

I'm now wrapping the code from main.rs in:
```

fn main() {

}
```

I get back:
```

error[E0254]: the name `chardet` is defined multiple times
 --> src/main.rs:3:5
  |
1 | extern crate chardet;
  | --------------------- previous import of the extern crate `chardet` here
2 | extern crate encoding;
3 | use chardet;
  |     ^^^^^^^ `chardet` reimported here
  |
  = note: `chardet` must be defined only once in the type namespace of this module

error[E0425]: cannot find value `filepath` in this scope
  --> src/main.rs:11:51
   |
11 |   let mut fh = OpenOptions::new().read(true).open(filepath).expect(
   |                                                   ^^^^^^^^ not found in this scope

error[E0425]: cannot find function `detect` in this scope
  --> src/main.rs:20:16
   |
20 |   let result = detect(&reader);
   |                ^^^^^^ not found in this scope
   |
help: possible candidate is found in another module, you can import it into scope
   |
3  | use chardet::detect;
   |

error[E0425]: cannot find function `charset2encoding` in this scope
  --> src/main.rs:26:42
   |
26 |   let coder = encoding_from_whatwg_label(charset2encoding(&result.0));
   |                                          ^^^^^^^^^^^^^^^^ not found in this scope
   |
help: possible candidate is found in another module, you can import it into scope
   |
3  | use chardet::charset2encoding;
   |

warning: unused import: `chardet`
 --> src/main.rs:3:5
  |
3 | use chardet;
  |     ^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default

error: aborting due to 4 previous errors

error: Could not compile `hello_world`.

```

--- a/README.md
+++ b/README.md
@@ -27,7 +27,8 @@ Using with encoding:
 ```rust
 extern crate chardet;
 extern crate encoding;
-use chardet;
+
+use chardet::{detect, charset2encoding};
 use std::fs::OpenOptions;
 use std::io::prelude::*;
 use encoding::DecoderTrap;

--- a/examples/detect.rs
+++ b/examples/detect.rs
@@ -1,0 +1,27 @@
+extern crate chardet;
+
+use std::fs::OpenOptions;
+use std::io::Read;
+
+fn main() {
+    let args: Vec<String> = std::env::args().collect();
+
+    if args.len() != 2 {
+        eprintln!("Usage: detect FILE");
+        std::process::exit(1);
+    }
+
+    // Open text file
+    let mut file = OpenOptions::new().read(true).open(&args[1])
+        .expect("Could not open file");
+
+    let mut reader: Vec<u8> = Vec::new();
+
+    file.read_to_end(&mut reader)
+        .expect("Could not read file");
+
+    // Detect charset of file
+    let result = chardet::detect(&reader);
+
+    println!("Detected {:?}", result);
+}

